src/iceberg/table_metadata.cc

```diff
@@ -408,6 +518,10 @@ class TableMetadataBuilder::Impl {
   const TableMetadata* base() const { return base_; }
   const TableMetadata& metadata() const { return metadata_; }
 
+  void SetLocation(std::string_view location) {
+    metadata_.location = std::string(location);
```

Comment:
We might need to follow the java impl to validate the input and then produce a update to the changes list.

```java
We might need to follow the Java impl to validate the input location and then produce a TableUpdate to the changes list.

```java
    public Builder setLocation(String newLocation) {
      if (location != null && location.equals(newLocation)) {
        return this;
      }

      this.location = newLocation;
      changes.add(new MetadataUpdate.SetLocation(newLocation));

      return this;
    }
```
```